plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -517,6 +521,36 @@ instance KnownBuiltinTypeIn DefaultUni term Integer => ReadKnownIn DefaultUni te
                  ]
     {-# INLINE readKnown #-}
 
+outOfBoundsErr :: Pretty a => a -> Vector.Vector term -> Text
+outOfBoundsErr x branches = fold
+    [ "'case "
+    , display x
+    , "' is out of bounds for the given number of branches: "
+    , display $ Vector.length branches
+    ]
+
+instance AnnotateCaseBuiltin DefaultUni where
+    annotateCaseBuiltin (SomeTypeIn uni) branches = case uni of
+        DefaultUniBool    -> Right $ map (, []) branches
+        DefaultUniInteger -> Right $ map (, []) branches
+        _                 -> Left $ display uni <> " isn't supported in 'case'"
+
+instance UniOf term ~ DefaultUni => CaseBuiltin term DefaultUni where
+    caseBuiltin (Some (ValueOf uni x)) branches = case uni of
+        DefaultUniBool -> case x of
+            -- We allow there to be only one branch as long as the scrutinee is 'False'.
+            -- This is strictly to save size by not having the 'True' branch if it was gonna be
+            -- 'Error' anyway.
+            False | len == 1 || len == 2 -> Right $ branches Vector.! 0
+            True  |             len == 2 -> Right $ branches Vector.! 1
+            _                            -> Left  $ outOfBoundsErr x branches
```

Comment:
Looks like casing on SOP allows extra branches. Should it also be allowed when casing on builtins?